### ISSUE TITLE
[Bug] Invalid conversion for rule "A 'substr()' call can behave as array access..."

### ISSUE THREAD

**Issue by kiler129** [kalessil/phpinspectionsea#980]
Take the following code:

```php
<?php

class foo {
    public function __toString() {
        return '123';
    }
}

class bar {
    public $test;

    public function __construct() {
        $this->test = new foo;
    }
}

$bar = new bar;
$char = \substr((string)$bar->test, 2, 1);

var_dump($char);
```
Demo: https://3v4l.org/bhQle

In such case PhpInspectionsEA Extended suggests replacing `substr` call:
![image](https://user-images.githubusercontent.com/1227834/42891394-e0bab87c-8a74-11e8-9227-09c42a68e94b.png)

By itself the suggestion is right, however the solution `$char = (string)$bar->test[2];` is invalid - it will throw "Cannot use object of type foo as array" (https://3v4l.org/IX0P6).

The correct solution for such case would be `$char = ((string)$bar->test)[2];` as shown at https://3v4l.org/kADHI

**Comment by kalessil** [comment]
Oops, thank you for reporting @kiler129, I'll take care about this bug.

**Comment by kalessil** [comment]
Fixed!

**Comment by kiler129** [comment]
Thank you :)
